### PR TITLE
Do not always persist ChannelManager on channel_update messages

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Based on #965 as it fixes a bug which would make this PR useless.

If we receive a `channel_update` message for a channel unrelated to
our own, we shouldn't trigger a persistence of our
`ChannelManager`. This avoids significant persistence traffic during
initial node startup.

### CODE REVIEW

**Review by valentinewallace — COMMENTED** [review]
Unit test would be cool if it's easy, otherwise LGTM

File: lightning/src/ln/channelmanager.rs [hunk 1]
```diff
@@ -4865,6 +4874,53 @@ mod tests {
...
+
+		// We check that the channel info nodes have doesn't change too early, even though we try
+		// to connect messages with new values
+		chan.0.contents.fee_base_msat *= 2;
+		chan.1.contents.fee_base_msat *= 2;
+		let node_a_chan_info = nodes[0].node.list_channels()[0].clone();
+		let node_b_chan_info = nodes[1].node.list_channels()[0].clone();
+
+		// The first two nodes (which opened a channel) should now require fresh persistence
+		assert!(nodes[0].node.await_persistable_update_timeout(Duration::from_millis(1)));
```
Comment (valentinewallace) [inline lightning/src/ln/channelmanager.rs:4897]: Not sure if by design, but don't think this test covers the `DoPersist` case: https://i.imgur.com/7fAhVOh.png

Comment (TheBlueMatt) [reply lightning/src/ln/channelmanager.rs:4897]: I'm not sure what I'm looking at there, but if not that's a bug - `await_persistable_update_timeout` is supposed to return true only if we have a reason to persist.

File: lightning/src/ln/channelmanager.rs [hunk 2]
```diff
@@ -4865,6 +4874,53 @@ mod tests {
 			}
 		}
 	}
+
+	#[test]
+	fn test_notify_limits() {
```
Comment (valentinewallace) [inline lightning/src/ln/channelmanager.rs:4879]: Weirdly when I run this test, only `trace` logs print out. Here's my log code: https://i.imgur.com/W37yQmk.png and here's what's printing out: https://i.imgur.com/DhpuX7u.png. Can't repro from randomly selected tests on `main`, but haven't investigated further

Comment (TheBlueMatt) [reply lightning/src/ln/channelmanager.rs:4879]: See second commit at #976

File: lightning/src/ln/channelmanager.rs (hunk 1 above)
Comment (valentinewallace) [reply lightning/src/ln/channelmanager.rs:4897]: Just meant that the test doesn't cover the case where we *do* want to persist after handling a channel update

Comment (TheBlueMatt) [reply lightning/src/ln/channelmanager.rs:4897]: Right, yes, sorry. As I was writing the test I found another bug, so the test is finished in the last commit at #949.

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without diff:
```
$ git diff-tree -U1 b7d249ab 803da875
$
```

File: lightning/src/ln/channel.rs [hunk 3]
```diff
@@ -288,7 +288,7 @@ impl HTLCCandidate {
 }
 
 /// Information needed for constructing an invoice route hint for this channel.
-#[derive(Clone)]
+#[derive(Clone, Debug, PartialEq)]
```
Comment (jkczyz) [inline lightning/src/ln/channel.rs:291]: Instead of implementing `PartialEq`, test that `fee_base_msat` hasn't changed.

**Review by jkczyz — APPROVED** [review]

File: lightning/src/ln/channel.rs (hunk 3 above)
Comment (TheBlueMatt) [reply lightning/src/ln/channel.rs:291]: I think its generally a good thing to have either way - its a public struct and its reasonable for users to want to compare it?

**Pull request merged**